.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,137 @@
+name: E2E Tests
+on:
+  workflow_call:
+
+jobs:
+    e2e-test:
+        name: E2E Tests - ${{ matrix.provider }}
+        runs-on: ${{ matrix.runner }}
+        if: ${{ matrix.condition }}
+        concurrency:
```

Comment:
This concurrency makes all of the cloud tests run in the same job. If it was removed, then the jobs would run in parallel in separate jobs.